File: C++/프로그래머스/2/42746. 가장 큰 수/가장 큰 수.cpp
#include <algorithm>
#include <vector>

using namespace std;

string solution(vector<int> numbers) {
    string answer = "";
    sort(numbers.begin(), numbers.end(), [&](int a, int b)->bool {
        return to_string(a) + to_string(b) > to_string(b) + to_string(a);
    });
    
    for(auto& num : numbers) {
        answer += to_string(num);
    }
    return answer[0] == '0' ? "0" : answer;
}

File: Python3/프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
import functools

def solution(numbers):
    def compare(x, y):
        return 1 if int(x + y) < int(y + x) else -1
    numbers = [str(n) for n in numbers]
    numbers.sort(key=functools.cmp_to_key(compare))
    
    return ''.join(numbers) if numbers[0] != '0' else "0"